### PR TITLE
[TK-13]: Logic for create an account

### PR DESCRIPTION
**Pull request by AbdulsKhan1**
## Jira: [TK-13](https://shop-go.atlassian.net/browse/TK-13?atlOrigin=eyJpIjoiYTY1MjUyMDhhMjFmNDQ4YjgwMThjNTZhZDdiM2Y5MDYiLCJwIjoiaiJ9)

**Description**:
- Restrict the user to enter more than 6 letters in password, password should be secure entry text, should not be visible all the time & password must have a special character and a number and no emoji.
- Add tap to dismiss in the screen to dismiss the keyboard & this screen should be scrollable, so that keyboard doesn't cover the button and text.
- Keep the next button disabled until user fills all the information 
- The email must have a valid format such as "johndoe@gmail.com"

**How to test it?**
- run the app on simulator and fill all text field
- password required special character and a number 
- email required "@" and ".co"

### CODE REVIEW

File: Vape-X/CreateAccountController.swift [hunk 1]
```diff
@@ -3,25 +3,88 @@
...
+        if phoneNumberFeild.text?.count == 10 {
+            phoneNumberFeild.text = phoneNumberText.applyPatternOnNumbers(pattern: "(###)###-####", replacementCharacter: "#")
+            phoneNumberFeild.resignFirstResponder()
+            nextButtonOutlet.isEnabled = true
+        }
+    }
+}
+
+extension String {
+
```
Comment (basitalich) [inline Vape-X/CreateAccountController.swift:69]: Move this to a seperate file

**Review by basitalich — CHANGES_REQUESTED** [review]

File: Vape-X/CreateAccountController.swift [hunk 2]
```diff
@@ -3,25 +3,88 @@
...
 import UIKit
 
 class CreateAccountController: UIViewController {
-
+    
     @IBOutlet weak var usernameFeild: UITextField!
     @IBOutlet weak var emailFeild: UITextField!
     @IBOutlet weak var phoneNumberFeild: UITextField!
     @IBOutlet weak var passwardLabel: UILabel!
     @IBOutlet weak var passwordTextfield: UITextField!
```
Comment (haroonmaqsoodch) [inline Vape-X/CreateAccountController.swift:15]: It's better to write Textfield after the property name, also fix the spellings of password

**Review by haroonmaqsoodch — APPROVED** [review]

File: Vape-X/CreateAccountController.swift (hunk 1 above)
Comment (AbdulsKhan1) [reply Vape-X/CreateAccountController.swift]: fixed.

File: Vape-X/CreateAccountController.swift (hunk 2 above)
Comment (AbdulsKhan1) [reply Vape-X/CreateAccountController.swift]: fixed.

**Review by ammadahmed22 — CHANGES_REQUESTED** [review]
where is username validation

**Review by ammadahmed22 — COMMENTED** [review]
disable emojis

**Review by ammadahmed22 — APPROVED** [review]
lgtm

**Pull request closed without merging**